### ISSUE TITLE
Weird indentations if externally-formatted strings are used

### ISSUE THREAD

**Issue by MoSal** [phsym/prettytable-rs#46]
``` rust
#[macro_use] extern crate prettytable;
extern crate ansi_term;

use prettytable::Table;
use ansi_term::Colour::Yellow;

fn main() {
    // Okay
    let mut table = Table::new();
    table.add_row(row![ Fybl->"AAAAAAA", Fmbl->"BBB", Fgbl->"CCC"]);
    table.add_row(row![ Fybl->"AAAAAAA", Fmbl->"BBB", Fgbl->"CCC"]);
    table.add_row(row![ Fybl->"AAAAAAA", Fmbl->"BBB", Fgbl->"CCC"]);
    table.add_row(row![ Fybl->"Yellow Yellow Yellow" ]);
    table.printstd();

    // Messed up
    let mut table2 = Table::new();
    table2.add_row(row![ Fybl->"AAAAAAA", Fmbl->"BBB", Fgbl->"CCC"]);
    table2.add_row(row![ Fybl->"AAAAAAA", Fmbl->"BBB", Fgbl->"CCC"]);
    table2.add_row(row![ Fybl->"AAAAAAA", Fmbl->"BBB", Fgbl->"CCC"]);
    table2.add_row(row![ Yellow.bold().paint("Yellow Yellow Yellow") ]);
    table2.printstd();
}
```

In `table2`, the first column in the first 3 rows has extra paddings. Those paddings exist regardless of the alignment requested.

**Comment by phsym** [comment]
Hi,
Unfortunately this is a known issue.
The reason is that the "Yellow Yellow Yellow" string contains ANSI escape sequences which are included when computing the strings length (to get the column width). But those sequences have no length once printed in a terminal.
The solution would be to remove those sequences from the string before computing the length, somewhere around [there](/phsym/prettytable-rs/blob/master/src/cell.rs#L29)

**Comment by m4b** [comment]
I'm seeing this too in a crate with mixed colored and prettytable, would be nice if this was adapted to, taken care of somehow when the string is inserted into the cell.

also this crate is great :heart:

**Comment by liautaud** [comment]
Another way to fix this is to take ANSI escape sequences into account when computing the strings length, which I did in #79.

**Comment by phsym** [comment]
Resolved by #79 thanks to @liautaud
